api/scpca_portal/models/project.py

```diff
@@ -881,30 +872,35 @@ def load_libraries_metadata(self, samples_metadata: List[Dict]):
         )
 
         for updated_sample_metadata in updated_samples_metadata:
-
             sample_id = updated_sample_metadata["scpca_sample_id"]
             sample_dir = self.get_sample_input_data_dir(updated_sample_metadata["scpca_sample_id"])
             sample_cell_count_estimate = 0
             sample_seq_units = set()
             sample_technologies = set()
 
-            library_metadata_paths = sorted(
-                list(Path(sample_dir).glob("*_metadata.json"))
-                + list(Path(sample_dir).rglob("*_spatial/*_metadata.json"))
-            )
-            for filename_path in library_metadata_paths:
-                library_json = metadata_file.load_library_metadata(filename_path)
+            single_cell_metadata_paths = set(Path(sample_dir).glob("*_metadata.json"))
+            spatial_metadata_paths = set(Path(sample_dir).rglob("*_spatial/*_metadata.json"))
 
-                if "filtered_cell_count" in library_json:
-                    sample_cell_count_estimate += library_json["filtered_cell_count"]
+            library_metadata_paths = list(single_cell_metadata_paths | spatial_metadata_paths)
+            all_libraries_metadata = [
+                metadata_file.load_library_metadata(path) for path in library_metadata_paths
+            ]
 
-                sample_seq_units.add(library_json["seq_unit"].strip())
-                sample_technologies.add(library_json["technology"].strip())
-
-                if "spatial" in str(filename_path):
-                    libraries_metadata[Sample.Modalities.SPATIAL].append(library_json)
-                else:
-                    libraries_metadata[Sample.Modalities.SINGLE_CELL].append(library_json)
+            for library_metadata, library_path in zip(
+                all_libraries_metadata, library_metadata_paths
+            ):
```

Comment:
```suggestion
            all_libraries_metadata = []
            for metadata_path in library_metadata_paths:
                library_metadata = metadata_file.load_library_metadata(metadata_path)
                all_libraries_metadata.applend(library_metadata)
```

Reply:
now that we extracted the modality and format outside of bulk create lets remove the zip as well